#!/usr/bin/env python
PACKAGE = "hector_motion_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add("motion_detect_threshold", int_t, 0, "Threshold ", 35, 0, 255)
#gen.add("motion_detect_min_percept_size", double_t, 0, "Min Percept Size", 1, 0, 1000)
#gen.add("motion_detect_max_percept_size", double_t, 0, "Max Percept Size", 300, 0, 1000)
#gen.add("motion_detect_min_density", double_t, 0, "Density", 1, 0, 1000)
#gen.add("percept_class_id", str_t, 0, "The percept class string", "victim")

gen.add("motion_detect_detectionLimit", int_t, 0, "Maximum Number of Motions to detect",100, 0, 100);
gen.add("motion_detect_min_area", int_t, 0, "Minimal area of detected motions", 300, 0, 10000);
gen.add("motion_detect_max_area", int_t, 0, "Maximal area of detected motions", 5000, 0, 20000);

gen.add("motion_detect_erosion", int_t, 0, "Iterations for erosion on fgimg", 2, 0, 8);
gen.add("motion_detect_dilation", int_t, 0, "Iterations for dilation on fgimg", 3, 0, 8);
gen.add("motion_detect_closing", int_t, 0, "Iterations for closing on fgimg", 1, 0, 8);
gen.add("motion_detect_euclidean_distance", int_t, 0, "Euclidean", 20, 0, 200);

gen.add("motion_detect_reduction", double_t, 0, "reduction parameter", 0.9, 0, 1);
gen.add("motion_detect_weight", double_t, 0, "weight parameter", 0.3, 0, 50);

gen.add("motion_detect_min_area_blob", double_t, 0, "min area for blobs", 1000, 0, 10000);
gen.add("motion_detect_max_area_blob", double_t, 0, "max area for blobs", 100000, 0, 100000);
gen.add("motion_detect_min_circularity_blob", double_t, 0, "min circularity filter", 0.2, 0, 1);
gen.add("motion_detect_max_circularity_blob", double_t, 0, "max circularity filter", 1, 0, 1);
gen.add("motion_detect_kernelSize", int_t, 0, "kernelsize for blob morp", 10, 0, 30);
gen.add("motion_detect_blob_morph", int_t, 0, "blob morph iterations", 1, 0, 8);

gen.add("motion_detect_tolerance", int_t, 0, "gge", 100000, 0, 10000000);

exit(gen.generate(PACKAGE, "motion_detection", "MotionDetection"))
